testsuites/syncgateway/functional/tests/test_xattrs.py

```diff
@@ -1847,3 +1847,194 @@ def verify_doc_ids_in_sdk_get_multi(response, expected_number_docs, expected_ids
+    if not xattrs_enabled:
+        pytest.skip('XATTR tests require --xattrs flag')
+
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+    sg_admin_url = cluster_topology['sync_gateways'][0]['admin']
+    sg_url = cluster_topology['sync_gateways'][0]['public']
+
+    bucket_name = 'data-bucket'
+    cbs_url = cluster_topology['couchbase_servers'][0]
+    sg_db = 'db'
+
+    log_info('Num docs per client: {}'.format(number_docs_per_client))
+    log_info('Num updates per doc per client: {}'.format(number_updates_per_doc_per_client))
+
+    log_info('sg_conf: {}'.format(sg_conf))
+    log_info('sg_admin_url: {}'.format(sg_admin_url))
+    log_info('sg_url: {}'.format(sg_url))
+
+    cluster = Cluster(config=cluster_conf)
+    cluster.reset(sg_config_path=sg_conf)
+
+    sg_tracking_prop = 'sg_one_updates'
+    sdk_tracking_prop = 'sdk_one_updates'
+
+    # Create sg user
+    sg_client = MobileRestClient()
+    sg_client.create_user(url=sg_admin_url, db=sg_db, name='autotest', password='pass', channels=['shared'])
+    autouser_session = sg_client.create_session(url=sg_admin_url, db=sg_db, name='autotest', password='pass')
+
+    # Connect to server via SDK
+    cbs_ip = host_for_url(cbs_url)
+    sdk_client = Bucket('couchbase://{}/{}'.format(cbs_ip, bucket_name), password='password', timeout=SDK_TIMEOUT)
+
+    # Inject custom properties into doc template
+    def update_props():
+        return {
+            'updates': 0,
+            sg_tracking_prop: 0,
+            sdk_tracking_prop: 0
+        }
+
+    # Create / add docs to sync gateway
+    sg_docs = document.create_docs(
+        'sg_doc',
+        number_docs_per_client,
+        channels=['shared'],
+        prop_generator=update_props
+    )
+    sg_docs_resp = sg_client.add_bulk_docs(
+        url=sg_url,
+        db=sg_db,
+        docs=sg_docs,
+        auth=autouser_session
+    )
+
+    sg_doc_ids = [doc['_id'] for doc in sg_docs]
+    assert len(sg_docs_resp) == number_docs_per_client
+
+    sg_create_docs, errors = sg_client.get_bulk_docs(url=sg_url, db=sg_db, doc_ids=sg_doc_ids,
+                                                     auth=autouser_session)
+    assert len(errors) == 0
+    sg_create_doc = sg_create_docs[0]["_rev"]
+    log_info("Sg created  doc revision :{}".format(sg_create_doc))
+
+    # Update docs via SDK
+    sdk_docs = sdk_client.get_multi(sg_doc_ids)
+    assert len(sdk_docs.keys()) == number_docs_per_client
+    for doc_id, val in sdk_docs.items():
+        doc_body = val.value
+        doc_body["updated_by_sdk"] = True
+        sdk_client.upsert(doc_id, doc_body)
+
+    sdk_first_update_docs, errors = sg_client.get_bulk_docs(url=sg_url, db=sg_db, doc_ids=sg_doc_ids,
+                                                            auth=autouser_session)
+    assert len(errors) == 0
+    sdk_first_update_doc = sdk_first_update_docs[0]["_rev"]
+    log_info("Sdk first update doc {}".format(sdk_first_update_doc))
```

Comment:
I did not add intentionally as reason for this test to verify two branched revisions when offline updates happens, and branched revisions verification is done. I added right now on all places which you commented, so that it can be  extra verification